cmd/common/image/build.go

```diff
@@ -320,6 +374,21 @@ func buildInContainer(opts *cmdOpts, conf *buildFile) error {
 		return err
 	}
 
+	imgOpts := &oci.ImageOptions{
+		VerifyOptions: oci.VerifyOptions{
+			VerifyPublicKey: opts.verifyImage,
+			PublicKey:       opts.publicKey,
+		},
+		AuthOptions: oci.AuthOptions{
+			AuthFile: oci.DefaultAuthFile,
+		},
+	}
+
+	err = verifyBuilderImage(ctx, cli, opts.builderImage, imgOpts)
+	if err != nil {
+		return fmt.Errorf("verifying builder image: %w", err)
+	}
+
```

Comment:
Can this be moved to `ensureBuilderImage` to verify only once when the image is pulled by the first time?

Reply:
I understand it, however it's a bit weird to verify this image each time it's run, I think the normal workflow is to verify the image only when it's pulled, after that, we can trust it not gonna be changed on the system. 
The issue here is that the user could run "docker pull ..." by their own, and then our verification won't work...

I suppose there are not other alternatives than verifying it each time.

Reply:
If I read this correctly, verifying the builder image costs more than building the gadget. IMO it's too much. I'm wondering if we can implement this in a different way:
- Pull and verify the image once on the CI job 
- Document that the builder image is signed and that users could verify it manually if they want 

@alban any comments on this?

Reply:
> I am not sure what you meant by this, as the image is already signed in the CI.

I mean to verify the image when building the gadgets on the CI job. It's true that it's signed by one job, but others jobs using it should verify it as well.

> Despite the cost being high, I still think we should go for it, or at least verifying the image when pulled and having users use the image though digest.
Indeed, using a verified builder image ensures the gadget are built correctly and without "troubles" added to them.

I pretty much agree with all of this, but still seems off to me to have to verify the builder image each time a gadget  is built, spending a lot of cpu cycles for nothing :(. IMO we should try to verify the image only at pull time: If the users pull the image by themselves, then it's their responsibility to verify it. I'm not sure however if using digest is a mess and difficult to implement.

> Do you have numbers on time taken to build an image with WASM?

I've noticed it's much slower but I don't have numbers.

Reply:
> Digest corresponding to latest.

Currently we hardcode `latest` in 

https://github.com/inspektor-gadget/inspektor-gadget/blob/fd01fbcd4d7b95eb02caacbd4dfa421369845c16/cmd/common/image/build.go#L52

how would it work with digests? 

> Numbers would be welcomed, so we can take a better decision knowing all the elements.

I don't think the wasm stuff should be taken into consideration here, those are independent things and one shouldn't affect the other. Even if building the wasm module is expensive, it's not a reason to have an expensive step to verify the container image.

Reply:
Take into consideration that many (possibly the majority of) gadgets won't use wasm.

Reply:
> If users do not have the requirements to build them using --local and they do not already have the image, they would need to pull it, so connection would be required.

My understanding of the previous discussion was that the verification process required connectivity even if the container image was already in the Docker store. Was it incorrect?

> I do not understand:
> 
>  1. You can perfectly give the digest to --builder-image.
> 2. In this case, the verification should occur with ig?

You can pass a digest to `--builder-image` but at the moment, the semantic is to designate the image to use. It is not meant to say "I trust this digest, no need to verify the cryptographic signatures".